src/uproot/source/fsspec.py

```diff
@@ -97,17 +97,20 @@ def chunks(self, ranges, notifications):
         self._num_requests += 1
         self._num_requested_chunks += len(ranges)
         self._num_requested_bytes += sum(stop - start for start, stop in ranges)
-        data = self._fs.cat_ranges(
-            [self._file_path] * len(ranges),
-            [start for start, _ in ranges],
-            [stop for _, stop in ranges],
-        )
+
         chunks = []
-        for item, (start, stop) in zip(data, ranges):
-            future = uproot.source.futures.TrivialFuture(item)
+        for start, stop in ranges:
+            future = uproot.source.futures.Future(
+                self._fs.cat_file, (self._file_path, start, stop)
+            )
             chunk = uproot.source.chunk.Chunk(self, start, stop, future)
             uproot.source.chunk.notifier(chunk, notifications)()
             chunks.append(chunk)
+
+        # TODO: Am I using the futures API correctly? Should I use some executor / resource future instead?
+        for chunk in chunks:
+            chunk.future._run()
```

Comment:
This blocks until all `chunks` are full, which defeats the purpose of making them Futures in the above loop. The idea is that this function should return a list of `chunks` that might or might not be full (more likely not, if the server is slow).